components/brave_wallet_ui/common/hooks/use-pending-transaction.ts

```diff
@@ -6,96 +6,280 @@
+  const { account: txAccount } = useAccountQuery(
+    transactionInfo?.fromAddress || skipToken
+  )
 
   // custom hooks
   const { getBlockchainTokenInfo, getERC20Allowance } = useLib()
-  const parseTransaction = useTransactionParser(transactionsNetwork)
   const { findAssetPrice } = usePricing(transactionSpotPrices)
   const {
     onFindTokenInfoByContractAddress,
     foundTokenInfoByContractAddress
   } = useTokenInfo(getBlockchainTokenInfo, visibleTokens, fullTokenList, transactionsNetwork)
 
   // state
-  const [suggestedMaxPriorityFeeChoices, setSuggestedMaxPriorityFeeChoices] = React.useState<string[]>([
-    transactionGasEstimates?.slowMaxPriorityFeePerGas || '0',
-    transactionGasEstimates?.avgMaxPriorityFeePerGas || '0',
-    transactionGasEstimates?.fastMaxPriorityFeePerGas || '0'
-  ])
-  const [baseFeePerGas, setBaseFeePerGas] = React.useState<string>(transactionGasEstimates?.baseFeePerGas || '')
   const [erc20AllowanceResult, setERC20AllowanceResult] = React.useState<
     string | undefined
   >(undefined)
 
-  // computed state
-  const transactionDetails = transactionInfo ? parseTransaction(transactionInfo) : undefined
+  // tx detail & gas memos
+  const gasFee = React.useMemo(() => {
+    if (!transactionInfo) {
+      return ''
+    }
+
+    return txCoinType === BraveWallet.CoinType.SOL
+      ? solFeeEstimate ?? ''
+      : getTransactionGasFee(transactionInfo)
+  }, [transactionInfo, txCoinType, solFeeEstimate])
+
+  const transactionDetails = React.useMemo(() => {
+    return transactionInfo
+      ? parseTransactionWithPrices({
+          tx: transactionInfo,
+          accounts,
+          fullTokenList,
+          gasFee,
+          spotPrices,
+          userVisibleTokensList: visibleTokens,
+          transactionNetwork: transactionsNetwork
+        })
+      : undefined
+  }, [
+    transactionInfo,
+    accounts,
+    fullTokenList,
+    solFeeEstimate,
+    spotPrices,
+    visibleTokens,
+    transactionsNetwork,
+    gasFee
+  ])
+
+  // balance queries
+  const { data: nativeBalance } = useGetAccountTokenCurrentBalanceQuery(
+    txAccount && transactionsNetwork
+      ? {
+          account: txAccount,
+          token: {
+            chainId: transactionsNetwork.chainId,
+            contractAddress: '',
+            isErc721: false,
+            isNft: false,
+            tokenId: ''
+          }
+        }
+      : skipToken
+  )
+
+  const { data: transferTokenBalance } = useGetAccountTokenCurrentBalanceQuery(
+    txAccount && transactionDetails?.token
+      ? {
+          account: txAccount,
+          token: {
+            chainId: transactionDetails.token.chainId,
+            contractAddress: transactionDetails.token.contractAddress,
+            isErc721: transactionDetails.token.isErc721,
+            isNft: transactionDetails.token.isNft,
+            tokenId: transactionDetails.token.tokenId
+          }
```

Comment:
Why not just use `transactionDetails.token`? Same for `nativeBalance` and `sellTokenBalance`.